src/features/TakeNow/ui/ValidateTakeNowParams.tsx

```diff
@@ -0,0 +1,180 @@
+import { useCallback, useRef, useState } from 'react';
+
+import { useActivityByIdQuery } from '~/entities/activity';
+import { appletModel, useAppletByIdQuery } from '~/entities/applet';
+import { useSubjectQuery } from '~/entities/subject';
+import { useUserState } from '~/entities/user/model/hooks';
+import { useWorkspaceAppletRespondent, useWorkspaceRolesQuery } from '~/entities/workspace';
+import { WorkspaceRole } from '~/shared/api/types/workspace';
+import { useNotification } from '~/shared/ui';
+import Loader from '~/shared/ui/Loader';
+import { useCustomTranslation } from '~/shared/utils';
+import { ActivityGroups } from '~/widgets/ActivityGroups';
+
+type ValidateTakeNowParamsProps = {
+  appletId: string;
+  startActivityOrFlow: string;
+  subjectId: string;
+  respondentId: string;
+};
+
+const TAKE_NOW_ROLES: WorkspaceRole[] = ['super_admin', 'owner', 'manager'];
+
+function ValidateTakeNowParams({
```

Comment:
That's a good idea, and it could make the component a lot more readable. I tried to do something like this before and it ended up being pretty convoluted but it came out decently this time. Guess I was trying to do it too early. Thanks!